modules/setting/storage.go

```diff
@@ -153,31 +153,44 @@ func getStorage(rootCfg ConfigProvider, name, typ string, sec ConfigSection) (*S
 		return nil, fmt.Errorf("invalid storage type %q", targetType)
 	}
 
+	// extra config section will be read SERVE_DIRECT, PATH, MINIO_BASE_PATH, MINIO_BUCKET to override the targetsec when possible
+	extraConfigSec := sec
+	if extraConfigSec == nil {
+		extraConfigSec = storageNameSec
+	}
+
 	var storage Storage
 	storage.Type = StorageType(targetType)
 
 	switch targetType {
 	case string(LocalStorageType):
-		storage.Path = ConfigSectionKeyString(targetSec, "PATH", filepath.Join(AppDataPath, name))
-		if !filepath.IsAbs(storage.Path) {
-			storage.Path = filepath.Join(AppWorkPath, storage.Path)
+		targetPath := ConfigSectionKeyString(targetSec, "PATH", "")
+		if targetPath == "" {
+			targetPath = AppDataPath
+		} else if !filepath.IsAbs(targetPath) {
+			targetPath = filepath.Join(AppDataPath, targetPath)
+		}
+
+		if extraConfigSec == nil {
+			storage.Path = filepath.Join(targetPath, name)
+		} else {
+			storage.Path = ConfigSectionKeyString(extraConfigSec, "PATH", filepath.Join(targetPath, name))
+			if !filepath.IsAbs(storage.Path) {
+				storage.Path = filepath.Join(AppDataPath, storage.Path)
```

Comment:
Here it should base on targetPath but not AppDataPath, and it also needs a test ....

Reply:
Here `extraConfigSec != nil`, that means we will use the override configuration. I have added 2 tests for this.

Reply:
But some tests are not right.

![image](https://github.com/go-gitea/gitea/assets/2114189/ad45c04c-b8c9-4788-ab77-9c5158aa12b1)

Reply:
Tests PASS

Reply:
"tests pass" doesn't mean the test code is right. Why here the path is a relative path? I don't think it's right.

Reply:
I have added a line `AppDataPath = "/tmp/data"` to make it not a relative path. At non-test env, `AppDataPath` can be guaranteed as absolute path but it's not in the test env.